test/features/build_server/compiler_interface_test.dart

```diff
@@ -0,0 +1,319 @@
+      final compiling = VHook(0);
+      final linking = VHook(0);
+      final waiting = VHook(0);
+      final done = VHook<bool?>(null);
+
+      testReceive.listen((message) {
+        if (message is InterfaceMessage && message.name == 'stateUpdate') {
+          final msg = message.args as CompileStatusMessage;
+          switch (msg.status) {
+            case CompileStatus.compiling:
+              compiling.set(compiling.value + 1);
+              break;
+            case CompileStatus.linking:
+              linking.set(linking.value + 1);
+              break;
+            case CompileStatus.waiting:
+              waiting.set(waiting.value + 1);
+              break;
+            case CompileStatus.done:
+              done.set(true);
+              break;
+            case CompileStatus.failed:
+              break;
+          }
+        }
+      });
+
+      worker.didStartCompilation();
+      worker.isCompiling(0);
+      worker.isCompiling(0.5);
+      worker.isCompiling(1);
+      worker.didFinishCompilation();
+      worker.didStartLinking();
+      worker.didFinishLinking();
+      worker.done();
+
+      await done.awaitValue(Duration(milliseconds: 100));
+
+      done.expect(isTrue);
+      compiling.expect(equals(5));
+      linking.expect(equals(1));
+      waiting.expect(equals(1));
+    });
+
+    test('can be initiated once', () async {
+      final testReceive = ReceivePort();
+      final receivePort = ReceivePort();
+      final sendPort = testReceive.sendPort;
+      final testSend = receivePort.sendPort;
+
+      WorkerInterface.start(sendPort, receivePort);
+
+      int responses = 0;
+
+      testReceive.listen((message) {
+        if (message is InterfaceMessage && message.name == 'init') {
+          responses += 1;
+        }
+      });
+
+      testSend.send(InterfaceMessage('init', [fileMap, config1, cache]));
+      testSend.send(InterfaceMessage('init', [fileMap, config1, cache]));
+
+      await Future.delayed(Duration(milliseconds: 100));
+
+      expect(responses, equals(1));
+    });
+
+    test('requires init as first call', () async {
+      final testReceive = ReceivePort();
+      final receivePort = ReceivePort();
+      final sendPort = testReceive.sendPort;
+      final testSend = receivePort.sendPort;
+
+      WorkerInterface.start(sendPort, receivePort);
+
+      bool hasAnswer = false;
+
+      testReceive.listen((message) {
+        if (message is InterfaceMessage) {
+          hasAnswer = true;
+        }
+      });
+
+      testSend.send(InterfaceMessage('hasCapacity', null));
+
+      await Future.delayed(Duration(milliseconds: 100));
+
+      expect(hasAnswer, isFalse);
+    });
+
+    test('responds to all build requests', () async {
+      final testReceive = ReceivePort();
+      final receivePort = ReceivePort();
+      final sendPort = testReceive.sendPort;
+      final testSend = receivePort.sendPort;
+      final buildCompleter = Completer<void>();
```

Comment:
Why not VHook? If you need the error behavior, I can integrate that.

Reply:
Yeah, but it isn't integrated. That's why not `VHook`

Reply:
Same thing here. [https://github.com/Coronon/flyde/pull/9#issuecomment-914277597](https://github.com/Coronon/flyde/pull/9#issuecomment-914277597)